### ISSUE TITLE
Remove array-find-index-polyfill?

### ISSUE THREAD

**Issue by fritx** [erikras/redux-form#2337]
I've upgraded redux-form from v3 to v6, a weird thing happened.

<img width="300" src="https://cloud.githubusercontent.com/assets/6647633/21477534/b42229a6-cb7d-11e6-8d0e-388d3d6cb5e4.png">

I realized it is redux-form that includes the polyfill.

It is fine in most cases, but it did cause an unexpected bug due to some bad code in our project:

```js
for (var key in arr) { } // wrong
for (var item in arr) { } // right
```

Just a suggestion, let's remove the (aggressive) polyfill? 😆  ...since it is only used in one place.

![image](https://cloud.githubusercontent.com/assets/6647633/21477631/893efab0-cb7e-11e6-9f3f-94920387f82f.png)

**Comment by gustavohenke** [comment]
I agree that we can remove that polyfill.

...but the code you're having problems with is exactly the same in both cases, just variables renamed :)

**Comment by fritx** [comment]
Great! @erikras

**Comment by fritx** [comment]
@gustavohenke ugh.. typo.. I meant `for item of arr`, updated!

**Comment by erikras** [comment]
Released as [`v6.5.0`](https://github.com/erikras/redux-form/releases/tag/v6.5.0).
